lib/aws.js

```diff
@@ -33,8 +33,7 @@ AWS.util.update(AWS.Config.prototype.keys, {
     var credentials = null;
     new AWS.CredentialProviderChain([
       function () { return new AWS.EnvironmentCredentials('AWS'); },
-      function () { return new AWS.EnvironmentCredentials('AMAZON'); },
-      function () { return new AWS.SharedIniFileCredentials(); }
```

Comment:
You could leave it in place, but it results in duplicate role assumptions. The reason being that the "resolve" here expects to complete synchronously and the role assumption completes asynchronously. So this method ends up returning null for assume-role profiles and the role has to be assumed again later.